spec/facter/resolvers/linux/networking_spec.rb

```diff
@@ -292,6 +294,8 @@
         allow(Facter::Util::Resolvers::Networking::PrimaryInterface).to receive(:read_from_ip_route).and_return(nil)
         allow(Facter::Util::Linux::Dhcp).to receive(:dhcp).with('lo', '1', log_spy).and_return(nil)
         allow(Facter::Util::Linux::Dhcp).to receive(:dhcp).with('ens160', '2', log_spy).and_return(nil)
+        allow(File).to receive(:exist?).with('/sys/class/net/lo/device').and_return(false)
+        allow(File).to receive(:exist?).with('/sys/class/net/ens160/device').and_return(true)
```

Comment:
These stubs need to go in the outermost `before :each` block at the top that way they will be evaluated for all tests in this file.

You'll want to add `allow(File).to receive(:exist?).and_call_original` after those two lines so that if `File.exists?` is called with other arguments, then it will still work.

Once you do that run the tests:

```
bundle exec rspec spec/facter/resolvers/linux/networking_spec.rb
```

If there are failures, you'll probably need to update some of the expectations (like the other PR), since facter will return the `physical` attribute for each interface.

Once you get all of these tests running, then run `bundle exec rake spec` to ensure all tests are passing and force push to your feature branch.

Reply:
I get about 10,000 lines of error messages trying to run `bundle install` on my fedora 39 system...